### PR TITLE
Upgrade Junction to Python 3.10 & Django 3.2 along with Social Login Fix

### PR DESCRIPTION
**Pull request by Bhandari423**
This upgrade is going to be incremental, this PR fixes initial syntactical issues, dependencies versions and few features which are not present. The idea is to fix things incrementally in a separate branch and then once everything is stable and we perform regression testing and raise a final PR towards the main.

This merge request contains changes as per Python 3.10:
- Python version upgrade to 3.10 and Django version to 3.2
- Replaced/Upgraded obsolete and deprecated libraries.
- Syntax changes as per Python 3.10 and Django 3.2
- Configurations for Sign In through social accounts like Gmail and GitHub.
- Configurations for Sign Up.

### CODE REVIEW

File: junction/base/utils.py [hunk 1]
```diff
@@ -3,13 +3,13 @@
 
 def get_date_diff_display(start, end):
     if end.year != start.year:
-        return "%s - %s" % (start.strftime("%-d %b %Y"), end.strftime("%-d %b %Y"))
+        return "%s - %s" % (start.strftime("%d %b %Y"), end.strftime("%d %b %Y"))
```
Comment (inovizz) [inline junction/base/utils.py:6]: Can we use f-strings here, please?

File: requirements.txt [hunk 2]
```diff
@@ -1,64 +1,54 @@
...
 
-# Ticket
-qrcode==5.1
-jsonfield==1.0.3
+
+#server
+#uwsgi
+
+# ssl certificate
+django-sslserver
```
Comment (inovizz) [inline requirements.txt:54]: can we pin this to a version number?

File: settings/common.py [hunk 3]
```diff
@@ -34,19 +37,18 @@
...
     "django.contrib.auth.middleware.AuthenticationMiddleware",
-    "django.contrib.auth.middleware.SessionAuthenticationMiddleware",
     "django.contrib.messages.middleware.MessageMiddleware",
     "django.middleware.clickjacking.XFrameOptionsMiddleware",
     "simple_history.middleware.HistoryRequestMiddleware",
 )
 
 CORE_APPS = (
-    "flat",  # https://github.com/elky/django-flat-theme
+    # "flat",  # https://github.com/elky/django-flat-theme
```
Comment (inovizz) [inline settings/common.py:51]: Could you please advise why we are commenting out this line?

**Review by inovizz — COMMENTED** [review]
First of all great work @Bhandari423 and @rohitchopra-epam. Most of the changes are good, except for some small changes and clarifications.

File: .gitignore [hunk 4]
```diff
@@ -91,3 +91,4 @@ qr_files/
 .vscode/
 
 tmp/
+sign_in_steps.txt
```
[the remark this answers is not in the document]
Comment (Bhandari423) [reply .gitignore]: fixed!

File: requirements.txt (hunk 2 above)
Comment (Bhandari423) [reply requirements.txt]: version number added!

File: settings/common.py (hunk 3 above)
Comment (Bhandari423) [reply settings/common.py:51]: django flat theme is for older version of django.
image

File: junction/base/utils.py (hunk 1 above)
Comment (Bhandari423) [reply junction/base/utils.py]: yes, we can. 
image

changes done!

**Review by inovizz — APPROVED** [review]

**Pull request merged**